behave/features/steps/general.py

```diff
@@ -9,17 +9,44 @@
 from selenium.webdriver.support.select import Select
 from datetime import date
 
+import re
+
 _BASE_URL = os.environ["WEB_APP_BASE_URL"]
 _DEFAULT_TIMEOUT = 30  # in seconds
 _POSTCODE_TIER_OVERRIDE = json.loads(os.environ["POSTCODE_TIER_OVERRIDE"])
 
 
+_NHS_BASE_URL = os.environ["NHS_BASE_URL"]
+_NHS_EMAIL = os.environ["NHS_EMAIL"]
+_NHS_PASSWORD = os.environ["NHS_PASSWORD"]
+_NHS_OTP = os.environ["NHS_OTP"]
+
+
+def get_url_match(url_pattern):
+    def url_match(c):
+        return re.match(url_pattern, c.current_url)
+
+    return url_match
+
+
 @then('I am redirected to the "{next_page_url}" page')
 def wait_until_url_present(context, next_page_url):
     WebDriverWait(context.browser, _DEFAULT_TIMEOUT).until(
         expected_conditions.url_to_be(f"{_BASE_URL}/{next_page_url}"))
 
 
+@then('I am redirected to a page that matches "{next_page_url}"')
+def wait_until_url_match(context, next_page_url):
+    WebDriverWait(context.browser, _DEFAULT_TIMEOUT).until(get_url_match(
```

Comment:
I'm not familiar with the either of the functions, but why is get_url_match used here and expected_conditions.url_to_be elsewhere?

Reply:
Because they do full matches on the url. Some of the new pages have params in the url, so we need to do regular expression matching on them

Reply:
cool, thanks for the explanation